MennaMohie/assignment3/Solution.cpp

```diff
@@ -66,107 +95,171 @@ class Trie
 		}
 	}
 
-	bool isWord(string word, checkType type = wordCheck)
+	//I need help doing the copy constructor :/
```

Comment:
Well, I actually don't think you actually need a copy constructor. I mean, it may be useful to be able to do a (deep) copy of Trie, but it's well beyond the current problem. The Finder class can keep a pointer to a Trie, instead of a copy of it (not the best design, it'll be a bit better to have a unique_ptr, but don't get into that if you don't want :) ) 
To be sure the compiler is not trying to copy nodes, you can add:
Trie(const Trie&) = delete;
Trie& operator =(const Trie&) = delete;
If you do want to do a copy, you need a deep one, i.e, you need to recursively make a copy of every node in the Trie, keeping the structure, of course.